beacon-chain/sync/initial-sync/blocks_fetcher.go

```diff
@@ -357,22 +358,31 @@ func (f *blocksFetcher) requestBlocks(
 		}
 	}()
 
-	resp := make([]*eth.SignedBeaconBlock, 0, req.Count)
+	blocks := make([]*eth.SignedBeaconBlock, 0, req.Count)
 	for i := uint64(0); ; i++ {
 		isFirstChunk := i == 0
 		blk, err := prysmsync.ReadChunkedBlock(stream, f.p2p, isFirstChunk)
 		if err == io.EOF {
 			break
 		}
-		// exit if more than max request blocks are returned
+		// Exit if more than max request blocks are returned.
 		if i >= params.BeaconNetworkConfig().MaxRequestBlocks {
 			break
 		}
 		if err != nil {
 			return nil, err
 		}
-		resp = append(resp, blk)
+		// Returned blocks MUST be in the slot range [start_slot, start_slot + count * step).
```

Comment:
Also lets harden the processing of incoming block's slot by their step. Ex:
previous block  was  slot 18, next block has to be 
`(x - 18) > 0 && (x-18) % 2 == 0`

Need to make sure incoming blocks are strictly increasing by their slot.

Reply:
Ok, will ensure both order and step increment.